### ISSUE TITLE
Result "errors(full: true).to_h" method output not expected

### ISSUE THREAD

**Issue by luccasmaso** [dry-rb/dry-validation#618]
The expected value for `Dry::Validation::Result` instance method `errors` is to  
- errors(full: true|false) (Boolean) — Get messages that include key names
- https://rubydoc.info/gems/dry-validation/Dry/Validation/Result#errors-instance_method

The current behavior is that, for `params do` the key is added to the message, but for `rule do` that adds a failure to the key it isn't:

dry-validation: 1.4

```
class User::Contract < Dry::Validation::Contract
  params do
    required(:pin).value(:string, size?: 4)
  end

  rule(:pin) do
    key.failure('must be only integer') if value.scan(/\D/).present?
  end
end
```

```
contract = Validators::User::Contract.new
contract.call(pin: "12345").errors(full: true).to_h
# {:pin=>["pin length must be 4"]}
contract.call(pin: "abcd").errors(full: true).to_h
# {:pin=>["must be only integer"]}
# Should be {:pin=>["pin must be only integer"]}
```

**Comment by korrs** [comment]
I started to fix this bug and found strange behaviour: When validation fails in params and rule definitions, then only error from params will be shown, error message from rule will be skipped.

For example:
```ruby
contract.call(pin: "a").errors(full: true).to_h # two errors, less than 4 and not digit
# {:pin=>["pin length must be 4"]}
```

But when params check passed we can see rule error
```ruby
contract.call(pin: "abcd").errors(full: true).to_h
# {:pin=>["must be only integer"]}
```

Is it expected behaviour?

I suppose, all error messages should be shown after validation:
```ruby
contract.call(pin: "a").errors(full: true).to_h
# {:pin=>["pin length must be 4", "pin must be only integer"]}
```

**Comment by flash-gordon** [comment]
@korrs absolutely, you cannot run rule checks on ill-typed input

**Comment by korrs** [comment]
👌

**Comment by sirfilip** [comment]
Hello love dryrb and love to help with this one,

Will create pr with proposed solution

**Comment by solnic** [comment]
This was fixed in 1.5.1
